tests/storage_tests.cpp

```diff
@@ -4,6 +4,49 @@
 
 using namespace sqlite_orm;
 
+#ifdef SQLITE_ORM_CTAD_SUPPORTED
+TEST_CASE("connection control") {
+    const auto openForever = GENERATE(false, true);
+    SECTION("") {
+        bool onOpenCalled = false;
```

Comment:
I'd change `bool` to `int` to cover amount of `onOpen` calls count just in case. I know your code doesn't produce such a bug but if we cover it with small amount of energy then it will be a win

Reply:
thx! You just saved my butt :)